src/chain/chain.rs

```diff
@@ -324,33 +357,10 @@ impl Chain {
     /// Removes an elder from our section, creating a new `our_info` in the process.
     /// Should not be called while a pfx change is in progress.
     pub fn remove_elder(&mut self, pub_id: PublicId) -> Result<EldersInfo, RoutingError> {
-        if self.state.change != PrefixChange::None {
-            log_or_panic!(
-                LogLevel::Warn,
-                "Removing {:?} from chain during pfx change.",
-                pub_id
-            );
-        }
-
-        if !self.our_prefix().matches(&pub_id.name()) {
-            log_or_panic!(
-                LogLevel::Error,
-                "Invalid RemoveElder event {:?} for self prefix.",
-                pub_id
-            );
-        }
+        self.assert_no_prefix_change("remove elder");
 
-        match self.state.our_members.get_mut(&pub_id) {
-            Some(info) => {
-                info.state = MemberState::Left;
-            }
-            None => {
-                log_or_panic!(
-                    LogLevel::Error,
-                    "Removed elder not {} present in our_members",
-                    pub_id
-                );
-            }
+        if let Some(info) = self.state.our_members.get_mut(&pub_id) {
+            info.persona = MemberPersona::Adult;
```

Comment:
Why do we convert it to an adult ?

Reply:
Because we are not removing the peer from the section, only making it non-elder. This will likely change with node ageing where we will have more granular transitions - `PromoteAdultToElder`, `PromoteInfantToElder`, `PromoteInfantToAdult`, `DemoteElderToAdult`, `DemoteElderToInfant`, `DemoteAdultToInfant`.